src/models/room.ts

```diff
@@ -785,6 +784,32 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         }
     }
 
+    /**
+     * Returns the last live event of this room.
+     * "last" means latest timestamp.
+     * "live of this room" means from all live timelines: the room and the threads.
+     *
+     * @returns MatrixEvent if there is a last event; else undefined.
+     */
+    public getLastLiveEvent(): MatrixEvent | undefined {
+        const roomEvents = this.getLiveTimeline().getEvents();
+        const lastRoomEvent = roomEvents[roomEvents.length - 1] as MatrixEvent | undefined;
+
+        return this.getThreads().reduce<MatrixEvent | undefined>(
+            (lastEventSoFar: MatrixEvent | undefined, thread: Thread) => {
+                const lastThreadEvent = thread.events[thread.events.length - 1];
+
+                if (!lastEventSoFar || (lastThreadEvent?.getTs() ?? 0) > (lastEventSoFar?.getTs() ?? 0)) {
```

Comment:
Please can you add a comment here and everywhere else you know of that uses `ts` for event ordering, referring to https://github.com/matrix-org/matrix-js-sdk/issues/3325 ?

Reply:
Done